src/notation/view/widgets/timelineview.cpp

```diff
@@ -103,6 +104,36 @@ using namespace mu::notation;
 TimelineView::TimelineView(QQuickItem* parent)
     : WidgetView(parent), muse::Injectable(muse::iocCtxForQmlObject(this))
 {
+    m_drawTimer.setSingleShot(true);
+    m_drawTimer.setInterval(32); // 30 fps
+
+    connect(&m_drawTimer, &QTimer::timeout, this, &TimelineView::doDraw);
+
+    doDraw();
+}
+
+void TimelineView::doDraw()
+{
+    m_image = QImage(width(), height(), QImage::Format_ARGB32_Premultiplied);
+    m_image.fill(Qt::transparent);
+
+    QPainter p;
+    p.begin(&m_image);
+
+    if (qWidget()) {
+        qWidget()->render(&p, QPoint(), QRegion(), QWidget::DrawWindowBackground | QWidget::DrawChildren);
```

Comment:
There is an overload that takes a QPaintDevice instead of QPainter. That might be more efficient, I don't really know.
```suggestion
        qWidget()->render(&image, QPoint(), QRegion(), QWidget::DrawWindowBackground | QWidget::DrawChildren);
```